hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/HeapData.java

```diff
@@ -63,14 +67,15 @@ public int totalSize() {
     @Override
     public int getPartitionHash() {
         if (hasPartitionHash()) {
-            return Bits.readIntB(payload, payload.length - Bits.INT_SIZE_IN_BYTES);
+            return Bits.readIntB(payload, payload.length - INT_SIZE_IN_BYTES);
         }
         return hashCode();
     }
 
     @Override
     public boolean hasPartitionHash() {
-        return totalSize() != 0 && payload[PARTITION_HASH_BIT_OFFSET] != 0;
+        return payload != null && payload.length > INT_SIZE_IN_BYTES
```

Comment:
Should we have payload.length >= HEAP_DATA_OVERHEAD instead of payload.length > INT_SIZE_IN_BYTES?